compiler/scopelang/dependency.ml

```diff
@@ -147,6 +149,8 @@ let build_program_dep_graph (prgm : 'm Ast.program) : SDependencies.t =
             used_defs g)
         g scope.Ast.scope_decl_rules)
     prgm.program_scopes g
+(* TODO FIXME: Add submodules here, they may still need dependency resolution
```

Comment:
So normally I would say we work under the assumption that dependent modules have already been typechecked and the typechecking is OK. So no need to redo the checks here.